wal/live_reader.go

```diff
@@ -27,26 +27,31 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+type LiveReaderMetrics struct {
+	ReaderCorruptionErrors *prometheus.CounterVec
+}
+
+func NewLiveReaderMetrics() *LiveReaderMetrics {
+	return &LiveReaderMetrics{
+		ReaderCorruptionErrors: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "prometheus_tsdb_wal_reader_corruption_errors_total",
+			Help: "Errors encountered when reading the WAL.",
+		}, []string{"error"}),
+	}
+}
+
 // NewLiveReader returns a new live reader.
-func NewLiveReader(logger log.Logger, reg prometheus.Registerer, r io.Reader) *LiveReader {
+func NewLiveReader(logger log.Logger, metrics *LiveReaderMetrics, r io.Reader) *LiveReader {
```

Comment:
For one metric I would just pass one metric, but up to you (:

Reply:
Personally I like the struct pattern a bit more in order to make it easy to add future metrics and not change the interface.

Reply:
This typically grows, I usually like to put this infrastructure in place right away, but no strong opinion.

Reply:
Not entirely sold, but happy to proceed as we are in hurry. (:

Reply:
I don't think we should skip, just because we're in a rush. Worst case we delay the release, that's fine. Maybe @brian-brazil can give his opinion.

Reply:
Right now I'd focus on getting the release out, I'd generally go with struct though.